src/templates/Article.jsx

```diff
@@ -5,11 +5,54 @@ import Helmet from 'react-helmet';
 
 import Breadcrumbs from '../templateComponents/Breadcrumbs.jsx';
 
+import titleify from '../../utils/titleify';
+
 const propTypes = {
   data: PropTypes.object,
   location: PropTypes.object
 };
 
+function getOgTitle(pathname) {
+  const pathMap = pathname
+    // remove leading and trailing slash
+    .replace(/^\/([a-z0-9/-]+[^/])\/?$/i, '$1')
+    .split('/');
+
+  let metaTitle = '';
+  pathMap.forEach((title, i) => {
+    if (i === pathMap.length - 1) {
+      metaTitle += titleify(title);
+    } else {
+      metaTitle += titleify(title + ' | ');
+    }
+  });
+
+  return metaTitle;
+}
+
+function getOgDescription(html) {
+  const el = document.createElement('html');
+  el.innerHTML = html;
+  let description = '';
+  const paragraph = el.getElementsByTagName('p')[0];
+  if (paragraph) {
+    description += paragraph.innerText;
+  }
+  return description;
+}
+
+function getOgImage(html) {
+  const el = document.createElement('html');
```

Comment:
I would query `markdownRemark` for this in graphql. `remark` converts the markdown file in to an AST. This might still be hanging around